torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2982,7 +2980,22 @@ def _gather(shape, index_dim, max_indices):
         (_tensor(()), (0, zero.clone().detach(), 2.5)),
     )
 
-    return [SampleInput(tensor, args=args) for tensor, args in test_cases]
+    samples = []
+    for tensor, args in test_cases:
+        samples.append(SampleInput(tensor, args=args))
+
+        if not requires_grad:
```

Comment:
Interesting! So these args are only supported when the input doesn't require grad?

Reply:
Yes. In fact, there were not _derivatives.yaml_ items for the reduce variants. I think I can implement them (not really confident about my math skills, though). As far as I understand, it could be implemented like:

- **add:** (same as `scatter_add`)
```python
> self_grad = grad
> src_grad = grad.gather(dim, index)
```
- **multiply:**
```python
> self_grad = grad.scatter(dim, index, src, 'multiply')
> src_grad = grad.gather(dim, index).mul(self.gather(dim, index))
```

What do you think?

Reply:
Yes sure that could be a different PR for sure.
Note that the current update is kind of problematic. So some update needs to be done to be sure we don't regress existing functions.